From 7bb547e46ed6236a3406b6800c5659489932ea21 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Fri, 3 Jun 2016 00:13:15 +0200
Subject: [PATCH 05/20] Add CPPFLAGS to CFLAGS when compiling C sources.
 Replace handcrafted build flags with /usr/share/ada/debian_packaging.mk: -
 Compile C sources with gcc instead of gnatgcc. - Handle
 DEB_BUILD_OPTIONS=parallel=N. - Avoid lintian warning about setting an
 explicit library runpath. - Make exact command lines available for automatic
 log checkers. - Tell gnatmake to echo commands to stdout instead of stderr,
 avoiding   buildds thinking it is inactive and killing it.

---
 debian/rules | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/debian/rules b/debian/rules
index 7530d52..7e3e584 100755
--- a/debian/rules
+++ b/debian/rules
@@ -2,12 +2,7 @@
 
 DPKG_EXPORT_BUILDFLAGS = 1
 include /usr/share/dpkg/buildflags.mk
-
-ADAFLAGS := $(filter-out -Wformat -Werror=format-security, $(CFLAGS))
-export ADAFLAGS
-
-CPUS := $(shell getconf _NPROCESSORS_ONLN)
-CC := gcc
+include /usr/share/ada/debian_packaging*.mk
 
 %:
 	dh ${@}
@@ -18,8 +13,8 @@ override_dh_auto_clean:
 
 override_dh_auto_build:
 	mkdir -p obj
-	$(CC) -c util-nl.c -o obj/util-nl.o $(CFLAGS)
-	gnatmake -j$(CPUS) -Pdebian/misc/adabrowse.gpr
+	$(CC) -c util-nl.c -o obj/util-nl.o $(CFLAGS) $(CPPFLAGS)
+	gnatmake $(BUILDER_OPTIONS) -Pdebian/misc/adabrowse.gpr
 
 override_dh_auto_test:
 override_dh_auto_install:
-- 
2.8.1

